app/themes/dashboard/ui/app/navigation/navigation.js

```diff
@@ -69,15 +72,24 @@
       return service;
     }
 
-    TopNavCtrl.$injector = ['$scope', 'userService'];
-    function TopNavCtrl($scope, userService) {
-      var ctrl = this;
+  TopNavCtrl.$injector = ['$scope', 'userService', '$document'];
 
-      ctrl.currentUser = userService.getUser();
+  function TopNavCtrl($scope, userService, $document) {
+    var ctrl = this;
 
-      $scope.$watch(userService.currentUser, function(newValue) {
-        ctrl.currentUser = newValue;
-      });
-    }
+    ctrl.currentUser = userService.getUser();
+
+    $scope.$watch(userService.currentUser, function(newValue) {
+      ctrl.currentUser = newValue;
+    });
+
+    ctrl.toggleSidebar = function() {
+      var body = angular.element($document[0].body);
+      var states = { collapsed: 'sidebar-collapse', open: 'sidebar-open' };
+      var collapsed = body.hasClass(states.collapsed);
+      body.removeClass(collapsed ? states.collapsed : states.open);
+      body.addClass(collapsed ? states.open : states.collapsed);
+    };
+  }
```

Comment:
Hmm, can't we use ng-show or ng-class for this purpose, and then just toggle the value of some boolean in ctrl or $scope?